src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -193,11 +200,31 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                         return;
                     }
 
-                    IEnumerable<ConfigurationClient> availableClients = _configClientManager.GetAvailableClients();
+                    IEnumerable<ConfigurationClient> clients = _configClientManager.GetClients();
+
+                    //
+                    // Filter clients based on their backoff status
+                    clients = clients.Where(client => 
+                    {
+                        Uri endpoint = _configClientManager.GetEndpointForClient(client);
+
+                        if (!_configClientBackoffs.TryGetValue(endpoint, out ConfigurationClientBackoffStatus clientBackoffStatus))
+                        {
+                            UpdateClientBackoffStatus(endpoint, true);
```

Comment:
There's no definition of success/fail here. I think it might be a bit better to just initialize it inline here and then add it into the dictionary.